### PR TITLE
Updated sample.yamls

### PR DESCRIPTION
**Pull request by HubertHugo**
Added builds that were in TAS

### CODE REVIEW

File: applications/matter_weather_station/sample.yaml [hunk 1]
```diff
@@ -2,8 +2,17 @@ sample:
...
     build_only: true
+    extra_args: DCMAKE_BUILD_TYPE=ZDebug CONFIG_CHIP_LIB_SHELL=y
+    platform_allow: thingy53_nrf5340_cpuapp
+    platform_exclude: thingy53_nrf5340_cpuapp_ns
+    integration_platforms:
+      - thingy53_nrf5340_cpuapp
+    tags: ci_build
+  applications.matter_weather_station.release:
+    build_only: true
+    extra_args: DCMAKE_BUILD_TYPE=ZRelease
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/sample.yaml:15]: Is `DCMAKE_BUILD_TYPE=ZRelease` used at all? Shouldn't we rather use `CONF_FILE=prj_release.conf`?

File: samples/matter/light_bulb/sample.yaml [hunk 2]
```diff
@@ -4,41 +4,85 @@ sample:
 tests:
   sample.matter.light_bulb.debug:
     build_only: true
+    extra_args: DCONFIG_CHIP_LIB_SHELL=y
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
-    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
```
Comment (Damian-Nordic) [inline samples/matter/light_bulb/sample.yaml:11]: nrf21540dk_nrf52840 doesn't have external flash and hence cannot be used in variants that have DFU enabled (all except `no_dfu`).

File: samples/matter/lock/sample.yaml [hunk 3]
```diff
@@ -4,33 +4,91 @@ sample:
...
+    extra_args: CONFIG_CHIP_LIB_SHELL=y CONFIG_CHIP_DFU_OVER_BT_SMP=y
+    integration_platforms:
+      - nrf52840dk_nrf52840
+      - nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
+    tags: ci_build
+  sample.matter.lock.release.smp_dfu:
+    build_only: true
+    extra_args: CONF_FILE=prj_release.conf CONFIG_CHIP_DFU_OVER_BT_SMP=y
     integration_platforms:
```
Comment (Damian-Nordic) [inline samples/matter/lock/sample.yaml:50]: IMHO it's not worth building all variants (debug + SMP DFU and release + SMP DFU) in the integration plan. Building only debug variants + a single release version should be enough to verify the build.

Comment (HubertHugo) [reply samples/matter/lock/sample.yaml:50]: I thin we're going to handle this another time, this is only unification of sample.yamls, and NCS.json in TAS. These builds are sum of builds in TAS and here.

**Pull request closed without merging**